docs/09_tutorials/01_binary-extension.md

```diff
@@ -0,0 +1,720 @@
+## eosio::binary_extension
+
+Let's fully explain what the `eosio::binary_extension` type is, what it does, and why we need it for contract upgrades in certain situations.
+
+You can find the implementation of `eosio::binary_extension` in the `eosio.cdt` repository in file: `eosio.cdt/libraries/eosiolib/core/eosio/binary_extension.hpp`.
```

Comment:
`...repository in [binary_extension.hpp](link to github source in develop branch)`